app/user-data-download/server/cronProcessDownloads.js

```diff
@@ -451,6 +454,22 @@ const generateUserAvatarFile = function(exportOperation, userData) {
 	}
 };
 
+function spliceByFileSize(fileList, size) {
+	let fileCounter = 0;
+	let sizeCounter = 0;
+
+	for (let i = 0; i < fileList.length; i++) {
+		fileCounter++;
+		sizeCounter += fileList[i].size;
+
+		if (sizeCounter > size) {
+			return fileList.splice(0, fileCounter);
+		}
+	}
+
+	return fileList.splice(0, fileList.length);
```

Comment:
Why do you need this 2nd splice here? 👀

Reply:
well noticed. i think this could simply return the original array